overlays/docker-compose.monitoring.yml

```diff
@@ -24,93 +24,98 @@ services:
       test: ["CMD", "curl", "-f", "http://host.docker.internal:3100/ready"]
       <<: *healthcheck-options
-
-  prometheus:
-    image: prom/prometheus:${PROMETHEUS_VERSION:-v2.45.0}
-    restart: unless-stopped
     user: nobody:nobody
-    read_only: true
-    security_opt:
-      - no-new-privileges:true
     cap_drop:
       - ALL
+    security_opt:
+      - no-new-privileges:true
+
+  # This is needed to avoid permission issues with the loki container when running as non-root user
+  loki-init:
+    image: busybox:1.37.0
+    volumes:
+      - loki:/loki
+    command:
+      [
+        "sh",
+        "-c",
+        "mkdir -p /loki/chunks /loki/rules /loki/wal /loki/boltdb-shipper-compactor && chown -R nobody:nobody /loki",
+      ]
+    user: root
+
+  prometheus:
+    image: prom/prometheus:${PROMETHEUS_VERSION:-v2.45.0}
     volumes:
       - prometheus:/prometheus
       - ./config/prometheus/prometheus.yml:/etc/prometheus/prometheus.yml:ro
     environment:
       PROMETHEUS_RETENTION_TIME: ${PROMETHEUS_RETENTION_TIME:-15d}
     command:
-      - '--config.file=/etc/prometheus/prometheus.yml'
-      - '--storage.tsdb.path=/prometheus'
-      - '--web.console.libraries=/etc/prometheus/console_libraries'
-      - '--web.console.templates=/etc/prometheus/consoles'
-      - '--storage.tsdb.retention.time=15d'
-      - '--web.enable-lifecycle'
-      - '--web.enable-admin-api'
+      - "--config.file=/etc/prometheus/prometheus.yml"
+      - "--storage.tsdb.path=/prometheus"
+      - "--web.console.libraries=/etc/prometheus/console_libraries"
+      - "--web.console.templates=/etc/prometheus/consoles"
+      - "--storage.tsdb.retention.time=15d"
+      - "--web.enable-lifecycle"
+      - "--web.enable-admin-api"
+    restart: unless-stopped
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:9090/-/healthy"]
       <<: *healthcheck-options
-
-  grafana:
-    image: grafana/grafana:${GRAFANA_VERSION:-10.0.0}
-    restart: unless-stopped
-    user: grafana:grafana
+    user: nobody:nobody
+    cap_drop:
+      - ALL
     read_only: true
     security_opt:
       - no-new-privileges:true
-    cap_drop:
-      - ALL
-    volumes:
-      - grafana:/var/lib/grafana
-      - ./config/grafana/provisioning:/etc/grafana/provisioning:ro
-    environment:
-      GF_SECURITY_ADMIN_PASSWORD: ${GRAFANA_ADMIN_PASSWORD}
-      GF_SECURITY_ADMIN_USER: admin
-      GF_USERS_ALLOW_SIGN_UP: false
-      GF_SECURITY_DISABLE_GRAVATAR: true
-      GF_ANALYTICS_REPORTING_ENABLED: false
-      GF_ANALYTICS_CHECK_FOR_UPDATES: false
-      GF_LOG_LEVEL: warn
-      GF_PATHS_PROVISIONING: /etc/grafana/provisioning
-      GF_SERVER_ROOT_URL: https://grafana.${HOSTNAME}
-      GF_SERVER_SERVE_FROM_SUB_PATH: false
-    healthcheck:
-      test: ["CMD", "curl", "-f", "http://localhost:3000/api/health"]
-      <<: *healthcheck-options
+
+  traefik:
+    logging: *loki-logging
 
 volumes:
-  loki: { }
-  prometheus: { }
-  grafana: { }
+  loki: {}
```

Comment:
Same as above, this formatting change is only related to the editor you're using